src/app/ride/CreateRide.js

```diff
@@ -0,0 +1,39 @@
+const Operation = require('../Operation');
+const Ride = require('../../domain/ride/Ride');
+
+class CreatRide extends Operation {
+  constructor({ ridesRepository, usersRepository }) {
+    super();
+    this.ridesRepository = ridesRepository;
+    this.usersRepository = usersRepository;
+  }
+
+  async execute({ driver, passengers, date }) {
+    const { SUCCESS, ERROR, VALIDATION_ERROR } = this.outputs;
+
+    const driverUser = await this.usersRepository.getUserByEmail(driver);
+
+    const { usersRepository } = this;
+    const passengerUserPromises = passengers.map(async (passengerEmail) => await usersRepository.getUserByEmail(passengerEmail));
+    const passengerUsers = await Promise.all(passengerUserPromises);
+
+    // Everything goes to repository must become Entity first
+    const ride = new Ride({ driver: driverUser, passengers: passengerUsers, date });
+
+    try {
+      const newRide = await this.ridesRepository.add(ride);
+
+      this.emit(SUCCESS, newRide);
+    } catch (error) {
+      if (error.message === 'ValidationError') {
+        return this.emit(VALIDATION_ERROR, error);
+      }
+
+      this.emit(ERROR, error);
+    }
+  }
+}
+
+CreatRide.setOutputs(['SUCCESS', 'ERROR', 'VALIDATION_ERROR']);

```

Comment:
I discover Symbol thanks to you so I don't know if my comment is pertinent. I saw in the link that Symbol is use for unique value (exemple with console.log). So why does not use Symbol here ?

Reply:
Great idea! Just pushed the fix. Can you please tell me if it resolves the conversation?